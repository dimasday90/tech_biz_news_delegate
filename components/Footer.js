import { Navbar, Nav } from "react-bootstrap";

export default function Footer() {
  return (
    <Navbar bg="dark" variant="dark" className="justify-content-center">
      <Navbar.Brand>
        Powered by{" "}
        <a href="https://newsapi.org" target="_blank">
          newsapi.org
        </a>
      </Navbar.Brand>
    </Navbar>
  );
}
