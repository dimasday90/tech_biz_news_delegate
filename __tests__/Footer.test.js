import React from "react";
import { render, cleanup } from "@testing-library/react";
import * as nextRouter from "next/router";
import renderer from "react-test-renderer";

import Footer from "../components/Footer";

afterEach(cleanup);

describe("Footer Test", () => {
  test("should render footer text", () => {
    nextRouter.useRouter = jest.fn();
    nextRouter.useRouter.mockImplementation(() => ({ route: "/" }));
    const { getByText } = render(<Footer />);
    expect(getByText("Powered by")).toBeInTheDocument();
    expect(getByText("newsapi.org")).toBeInTheDocument();
  });
});

describe("With Snapshot Testing", () => {
  it("should match footer coding", () => {
    nextRouter.useRouter = jest.fn();
    nextRouter.useRouter.mockImplementation(() => ({ route: "/" }));
    const component = renderer.create(<Footer />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
