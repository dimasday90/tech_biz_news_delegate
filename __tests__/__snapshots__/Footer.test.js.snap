// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing should match footer coding 1`] = `
<nav
  className="justify-content-center navbar navbar-expand navbar-dark bg-dark"
>
  <span
    className="navbar-brand"
  >
    Powered by
     
    <a
      href="https://newsapi.org"
      target="_blank"
    >
      newsapi.org
    </a>
  </span>
</nav>
`;
